### ISSUE TITLE
Crash on zext i1 to i16

### ISSUE THREAD

**Issue by krasin** [llvm-dcpu16/llvm-dcpu16#92]
See test/CodeGen/DCPU16/bit.ll

```
define i16 @bitwrr(i16 %a, i16 %b) nounwind {
    %t1 = and i16 %a, %b
    %t2 = icmp ne i16 %t1, 0
    %t3 = zext i1 %t2 to i16
    ret i16 %t3
}
```

llc crashes on zext with:

```
/disk2/llvm-dcpu16/cbuild/bin/./llc < /disk2/llvm-dcpu16/test/CodeGen/DCPU16/bit.ll -march=dcpu16 | /disk2/llvm-dcpu16/cbuild/bin/./FileCheck /disk2/llvm-dcpu16/test/CodeGen/DCPU16/bit.ll
--
Exit Code: 1
Command Output (stderr):
--
Impossible reg-to-reg copy
UNREACHABLE executed at /disk2/llvm-dcpu16/lib/Target/DCPU16/DCPU16InstrInfo.cpp:89!
Stack dump:
0.  Program arguments: /disk2/llvm-dcpu16/cbuild/bin/./llc -march=dcpu16 
1.  Running pass 'Function Pass Manager' on module '<stdin>'.
2.  Running pass 'Post-RA pseudo instruction expansion pass' on function '@bitwrr'
FileCheck error: '-' is empty.
--
```

This may or not be related to #81.

**Comment by krasin** [comment]
A bit more evidence. I have ran bin/llc under gdb:

```
gdb bin/llc --args bin/llc -march dcpu16 ~/bit.ll


UNREACHABLE executed at /disk2/llvm-dcpu16/lib/Target/DCPU16/DCPU16InstrInfo.cpp:89!

Program received signal SIGABRT, Aborted.
0x00007ffff6eaf3a5 in raise () from /lib/x86_64-linux-gnu/libc.so.6
(gdb) up
#1  0x00007ffff6eb2b0b in abort () from /lib/x86_64-linux-gnu/libc.so.6
(gdb) up
#2  0x0000000001367477 in llvm::llvm_unreachable_internal (msg=0x1498efa "Impossible reg-to-reg copy", file=0x1498e90 "/disk2/llvm-dcpu16/lib/Target/DCPU16/DCPU16InstrInfo.cpp", line=89)
    at /disk2/llvm-dcpu16/lib/Support/ErrorHandling.cpp:98
98    abort();
(gdb) up
#3  0x0000000000a38dbb in llvm::DCPU16InstrInfo::copyPhysReg (this=0x1dd0510, MBB=..., I=..., DL=..., DestReg=1, SrcReg=6, KillSrc=false) at /disk2/llvm-dcpu16/lib/Target/DCPU16/DCPU16InstrInfo.cpp:89
89      llvm_unreachable("Impossible reg-to-reg copy");
(gdb) l
84                                      bool KillSrc) const {
85            unsigned Opc;
86              if (DCPU16::GR16RegClass.contains(DestReg, SrcReg))
87                  Opc = DCPU16::MOV16rr;
88              else
89                  llvm_unreachable("Impossible reg-to-reg copy");
90
91                    BuildMI(MBB, I, DL, get(Opc), DestReg)
92                  .addReg(SrcReg, getKillRegState(KillSrc));
93            }
(gdb) p DestReg
$1 = 1
(gdb) p SrcReg
$2 = 6
```

Let's take a look, which registers are 1 and 6. It's in cbuild/lib/Target/DCPU16/DCPU16GenRegisterInfo.inc

```
namespace DCPU16 {
enum {
  NoRegister,
  RA = 1,
  RB = 2,
  RC = 3,
  RI = 4,
  RJ = 5,
  RO = 6,
  RSP = 7,
  RX = 8,
  RY = 9,
  RZ = 10,
  NUM_TARGET_REGS       // 11
};
}
```

Aha! 6 is RO which does not belong to GR16RegClass. So, it's likely to be a separate bug, not #81.

**Comment by ghost** [comment]
Fixed with d569253.
